cpp/src/arrow/util/compression_zstd.cc

```diff
@@ -180,12 +189,27 @@ Status ZSTDCompressor::End(int64_t output_len, uint8_t* output, int64_t* bytes_w
   *should_retry = ret > 0;
   return Status::OK();
 }
+} // namespace
 
 // ----------------------------------------------------------------------
 // ZSTD codec implementation
 
+ZSTDCodec::ZSTDCodec() : Codec(), zstdCompressionLevel_(kZSTDDefaultCompressionLevel) {
+  const char* envZstdCompressionLevel = getenv("ARROW_ZSTD_COMPRESSION_LEVEL");
+  if (envZstdCompressionLevel != NULLPTR) {
+    long int userSpecifiedLevel = strtol(envZstdCompressionLevel, NULLPTR, 10);
+    const bool hasErrorInParsing = (userSpecifiedLevel == 0L && errno == ERANGE);
+    constexpr long int intMin = static_cast<long int>(std::numeric_limits<int>::min());
+    constexpr long int intMax = static_cast<long int>(std::numeric_limits<int>::max());
+    const bool isOutOfIntBounds = ((userSpecifiedLevel < intMin) || (userSpecifiedLevel > intMax));
+    if (hasErrorInParsing == false && isOutOfIntBounds == false) {
+      zstdCompressionLevel_ = static_cast<int>(userSpecifiedLevel);
```

Comment:
Could you explain the various `static_cast`s being used here?  It looks like the compression level is destined to be an `int` type so I think you can do away with the up-cast then downcast like this:

```cpp
    constexpr int intMin = std::numeric_limits<int>::min());
    constexpr int intMax = std::numeric_limits<int>::max());
    const bool isOutOfIntBounds = ((userSpecifiedLevel < intMin) || (userSpecifiedLevel > intMax));
    if (hasErrorInParsing == false && isOutOfIntBounds == false) {
      zstdCompressionLevel_ = userSpecifiedLevel;
``` 

Not sure if I missed something though?

Reply:
No longer relevant since that patch has changed.